### DISCUSSION TITLE
Issues finding python nbd module after update of python

### DISCUSSION THREAD

**Discussion by rettw78** [abbbi/virtnbdbackup#144] [Q&A / Support] [answered]
Hi -

I have been running virtnbdbackup successfully as a backup solution for a few months now on a couple of Arch Linux based KVM hosts.  After a recent update virtnbdbackup now fails with the following error message:

/usr/bin/virtnbdbackup:4: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  __import__('pkg_resources').run_script('virtnbdbackup==1.9.45', 'virtnbdbackup')
Traceback (most recent call last):
  File "/usr/bin/virtnbdbackup", line 4, in <module>
    __import__('pkg_resources').run_script('virtnbdbackup==1.9.45', 'virtnbdbackup')
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 722, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python3.11/site-packages/pkg_resources/__init__.py", line 1572, in run_script
    exec(script_code, namespace, namespace)
  File "/usr/lib/python3.11/site-packages/virtnbdbackup-1.9.45-py3.11.egg/EGG-INFO/scripts/virtnbdbackup", line 33, in <module>
  File "/usr/lib/python3.11/site-packages/virtnbdbackup-1.9.45-py3.11.egg/libvirtnbdbackup/nbdcli/__init__.py", line 21, in <module>
  File "/usr/lib/python3.11/site-packages/virtnbdbackup-1.9.45-py3.11.egg/libvirtnbdbackup/nbdcli/client.py", line 21, in <module>
ModuleNotFoundError: No module named 'nbd'

I have made sure all dependencies on arch are installed as they were before the update (which included python 3.11.3-> 3.11.5) and I have been unable to fix the missing module.  Archlinux needs nbdkit installed from the AUR to provide nbd functions/plugins, and also tried the latest virtnbdbackup version.    Any help would greatly appreciated.

Let me know if you need additional information.  

Thanks,

Rett

**Comment by abbbi** [comment]
it seems you are missing the python libnbd bindings, im not using arch so i cant tell which package you would have to install (on debian, its python3-libnbd)

It not quite virtnbdbackup related, i guess if you simply do

`python3 -c "import nbd"`

you probably run into the same issue. Guess the new python version either comes without the module or you need to
setup the module for your new python 3.11.5 installation.

**Reply by rettw78** [reply, marked as answer]
Thanks for the reply. It appears that arch no longer supplies this package in any form that I can find.  I found that if I compile an old version of libnbd from git I can get the python bindings for nbd back, however this seems like a hacky way of doing things.  I will see if I can get someone on the Archlinux side to tell me why this is no longer available.
